util/cgiflash.c

```diff
@@ -244,26 +276,41 @@ CgiStatus ICACHE_FLASH_ATTR cgiUploadFirmware(HttpdConnData *connData) {
 	printf("state->len %d, state->address: %d\n", state->len, state->address);
 #endif
 
-	if (connData->post.len == connData->post.received) {
+	if  (connData->post.len == connData->post.received) {
+		cJSON *root = NULL;
+		char *json_string = NULL;
+		root = cJSON_CreateObject();
 		//We're done! Format a response.
-		ESP_LOGD(TAG, "Upload done. Sending response");
+		ESP_LOGI(TAG, "Upload done. Sending response");
 		httpdStartResponse(connData, state->state==FLST_ERROR?400:200);
-		httpdHeader(connData, "Content-Type", "text/plain");
+		httpdHeader(connData, "Content-Type", "application/json; charset=utf-8"); //We are going to send some JSON.
 		httpdEndHeaders(connData);
-		if (state->state!=FLST_DONE) {
-			httpdSend(connData, "Firmware image error:", -1);
-			httpdSend(connData, state->err, -1);
-			httpdSend(connData, "\n", -1);
-		} else {
+		if (state->state==FLST_DONE) {
 			if (esp_ota_end(state->update_handle) != ESP_OK) {
+				state->err="esp_ota_end failed!";
 		        ESP_LOGE(TAG, "esp_ota_end failed!");
+		        state->state=FLST_ERROR;
 		    }
-		    err = esp_ota_set_boot_partition(state->update_partition);
-		    if (err != ESP_OK) {
-		        ESP_LOGE(TAG, "esp_ota_set_boot_partition failed! err=0x%x", err);
-		    }
+			else
+			{
+				state->err="Flash Success.";
+			}
+//		    err = esp_ota_set_boot_partition(state->update_partition);
```

Comment:
can you drop these commented out lines?

Reply:
1.  That commented line is already changed in a later commit on this pull request.  
2.  That sounds like some advanced git-fu.  Would you please point me to some step-by-step instructions on how to achieve what you're asking?

Reply:
Hi @phatpaul. interactive rebasing is pretty common but agreed that it does seem advanced if you haven't done it a bit. https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d has some good ideas.

Also you should be able to remove the merge commits as a side effect of the rebase, the ideal situation is to have the commits be clean and the final version of what you'd like to accomplish. Effectively each commit that 'fixes' an earlier commit in the same commit series should in fact be a modification to the original commit to make it correct in the first place, I don't have to see the initial code + the fixes, just the final result of a given commit is ideal.

Reply:
@chmorgan thanks for the link.  But I'm afraid of rewriting git history, especially since I've already pushed these commits to github.
Are you suggesting that I change the history and then do a forced push to my github repo?  I think I'd rather create a new branch and copy these changes to it.